xonsh/proc.py

```diff
@@ -220,8 +220,8 @@ def __init__(self, fd, timeout=None):
 
 def populate_buffer(reader, fd, buffer, chunksize):
     """Reads bytes from the file descriptor and copies them into a buffer.
-    The reads happend in parallel, using pread(), and is thus only
-    availabe on posix. If the read fails for any reason, the reader is
+    The reads happened in parallel, using pread(), and is thus only
```

Comment:
I wasn't paying much attention to tense and grammar as I went through, just the full list of misspellings. That said, looking over the sentence now, it definitely could use some work. I do think your sentence is better all around. I can make this change as well.